dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -126,7 +136,7 @@ public Response replaceFileInDataset(
                     // (2b) Load up optional params via JSON
                     //  - Will skip extra attributes which includes fileToReplaceId and forceReplace
                     //---------------------------------------
-                    optionalFileParams = new OptionalFileParams(jsonData);
+                    optionalFileParams = new OptionalFileParams().create(jsonData);
```

Comment:
This bean won't work properly since you are initializing it with "new".

Reply:
forgot to refactor it back.